lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -116,35 +132,74 @@ def __init__(
         )
         # TODO Future PRs handle scale down
 
-    def _on_relation_changed(self, event):
+    def _on_relation_changed(self, event) -> None:
         """Starts/restarts monogs with config server information."""
         relation_data = event.relation.data[event.app]
-        if not relation_data.get(KEYFILE_KEY):
+        if not relation_data.get(KEYFILE_KEY) or not relation_data.get(CONFIG_SERVER_DB_KEY):
             event.defer()
             self.charm.unit.status = WaitingStatus("Waiting for secrets from config-server")
             return
 
-        self.update_keyfile(key_file_contents=relation_data.get(KEYFILE_KEY))
+        updated_keyfile = self.update_keyfile(key_file_contents=relation_data.get(KEYFILE_KEY))
+        updated_config = self.update_config_server_db(
+            config_server_db=relation_data.get(CONFIG_SERVER_DB_KEY)
+        )
+
+        # avoid restarting mongos when possible
+        if not updated_keyfile and not updated_config and self.charm.monogs_initialised:
```

Comment:
spell check `monogs_initialised `

Reply:
What do we want to check in `mongos_initialised` additionally? I don't remember anything that we can consider as mongos initialisation. it is or started or not started

Reply:
Hmm yes, we could check ` if not self.is_mongos_running():`, but this code has a retry in it, so it takes more time to run, perhaps I can disable the retry in certain cases